app/org/sagebionetworks/bridge/services/ParticipantService.java

```diff
@@ -261,6 +273,34 @@ public StudyParticipant getParticipant(Study study, Account account, boolean inc
         }
 
         StudyParticipant.Builder builder = new StudyParticipant.Builder();
+        SubstudyAssociations assoc = BridgeUtils.substudyAssociationsVisibleToCaller(account.getAccountSubstudies());
+        addAccount(builder, assoc, account);
+    
+        CriteriaContext.Builder contextBuilder = new CriteriaContext.Builder();
+        contextBuilder.withStudyIdentifier(study.getStudyIdentifier());
+        contextBuilder.withUserId(account.getId());
+        contextBuilder.withHealthCode(account.getHealthCode());
+        contextBuilder.withUserDataGroups(account.getDataGroups());
+        contextBuilder.withUserSubstudyIds(assoc.getSubstudyIdsVisibleToCaller());
```

Comment:
Since addHistory() doesn't actually need the context, would it be better to include this in the second if (requestInfo != null) below? This keeps the code simpler and reduces the amount of code churn.

Reply:
I moved this around so we only create the context if we have the RequestInfo to get the consent status.